File: tests/src/org.apache.htrace/htrace-core/3.1.0-incubating/src/test/java/org_apache_htrace/htrace_core/EnumResolverTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_htrace.htrace_core;

import org.apache.htrace.fasterxml.jackson.annotation.JsonValue;
import org.apache.htrace.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class EnumResolverTest {
    @Test
    void deserializesEnumUsingJsonValueMethod() throws Exception {
        ObjectMapper mapper = new ObjectMapper();

        SpanRelationship relationship = mapper.readValue("\"child-span\"", SpanRelationship.class);

        assertThat(relationship).isSameAs(SpanRelationship.CHILD);
        assertThat(mapper.readValue("\"root-span\"", SpanRelationship.class)).isSameAs(SpanRelationship.ROOT);
    }

    public enum SpanRelationship {
        ROOT("root-span"),
        CHILD("child-span");

        private final String jsonValue;

        SpanRelationship(String jsonValue) {
            this.jsonValue = jsonValue;
        }

        @JsonValue
        public String asJsonValue() {
            return jsonValue;
        }
    }
}

File: tests/src/com.fasterxml.jackson.module/jackson-module-scala_2.13/2.11.4/src/test/scala/com_fasterxml_jackson_module/jackson_module_scala_2_13/EnumResolverTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_fasterxml_jackson_module.jackson_module_scala_2_13

import com.fasterxml.jackson.module.scala.util.EnumResolver
import org.junit.jupiter.api.Assertions.{assertEquals, assertSame}
import org.junit.jupiter.api.Test

class EnumResolverTest {
  @Test
  def resolvesScalaEnumerationFromModuleClass(): Unit = {
    val resolver: EnumResolver = EnumResolver(EnumResolverFixture.getClass)

    assertSame(EnumResolverFixture.getClass, resolver.getEnumClass)
    assertEquals(EnumResolverFixture.Green, resolver.getEnum("green"))
  }
}

object EnumResolverFixture extends Enumeration {
  val Red: Value = Value("red")
  val Green: Value = Value("green")
  val Blue: Value = Value("blue")
}
